### ISSUE TITLE
ENH: add a warning in mask-extraction routines when the fail

### ISSUE THREAD

**Issue by GaelVaroquaux** [nilearn/nilearn#1276]
We should check the number of voxels in the mask at the end of a mask-extraction routine and issue a customized warning to the user if the number of voxels is either 0 or total size of the array.

The warning should be added at the end of compute_epi_mask, compute_multi_epi_mask, compute_background_mask, compute_multi_background_mask.

The warning should be explicit that "Automatic mask computation has failed on image %s". It should provide common problems and their solutions for instance: using compute_epi on images that are not EPI, or vice versa, or using compute_epi on images that were detrended, centered or variance normalized.

**Comment by man-shu** [comment]
I just ran into this one while working on this test for multi_pca:
https://github.com/nilearn/nilearn/blob/70af352e38936dddbdb959fe85262885ab09ea4d/nilearn/decomposition/tests/test_multi_pca.py#L148

When a mask is not provided, NiftiMasker promises to compute it, but when this computation fails (i.e. returns an empty mask), it raises the following error while loading the empty mask via 

https://github.com/nilearn/nilearn/blob/70af352e38936dddbdb959fe85262885ab09ea4d/nilearn/maskers/nifti_masker.py#L516

```bash
ValueError: The mask is invalid as it is empty: it masks all data.
```

I think it would be better to check if the computed mask is empty and then raise the error after the computation is done in functions like https://github.com/nilearn/nilearn/blob/70af352e38936dddbdb959fe85262885ab09ea4d/nilearn/masking.py#L263

**Comment by Remi-Gau** [comment]
pretty sure that this is what is happening:

mask gets computed a few lines above:

https://github.com/nilearn/nilearn/blob/70af352e38936dddbdb959fe85262885ab09ea4d/nilearn/maskers/nifti_masker.py#L469-L471

or are you saying that some warning or error should already be thrown in compute_epi_mask or similar functions?

**Comment by man-shu** [comment]
> or are you saying that some warning or error should already be thrown in compute_epi_mask or similar functions?

yes exactly! Currently, `load_mask_img` is throwing the empty mask error, and at least for a while, I was interpreting that to be the default when we pass `mask=None`. 

Broke my brain a bit because the docs say that when `mask=None` we will compute a mask automatically

**Comment by Remi-Gau** [comment]
as the title of this issue mention I think the compute mask should only throw warnings (though the warning could mention that this could lead to an error in the follow up steps)

**Comment by man-shu** [comment]
Ok, I just realized we already throw a warning when we compute an empty mask via this function:

https://github.com/nilearn/nilearn/blob/ed25d2925e1e4d9ba9c9c0381a4e72413bd69bd5/nilearn/masking.py#L236

```bash
_MaskWarning: Computed an empty mask. Are you sure that input data are EPI images not detrended. 
```

So shall we close this one?

**Comment by Remi-Gau** [comment]
is this the case for all the compute mask functions?

**Comment by man-shu** [comment]
Yes!

**Comment by Remi-Gau** [comment]
ok then I think this can be closed
